### PR TITLE
capture slotted elements

### PR DESCRIPTION
**Pull request by cWenyu**
Fix issue https://github.com/1904labs/dom-to-image-more/issues/178 </br>
Also fix a typo

### CODE REVIEW

File: src/dom-to-image-more.js [hunk 1]
```diff
@@ -97,8 +97,8 @@
         let restorations = [];
         return Promise.resolve(node)
             .then(ensureElement)
-            .then(function (clonee) {
-                return cloneNode(clonee, options, null, ownerWindow);
+            .then(function (clone) {
```
Comment (IDisposable) [inline src/dom-to-image-more.js:100]: This is NOT a typo, the intent here is to ensure that the `clone` in scope isn't hidden by this name.

Please revert these two lines

File: src/dom-to-image-more.js [hunk 2]
```diff
@@ -658,9 +664,13 @@
...
 
         function isInShadowRoot(value) {
+            // Object.prototype.hasOwnProperty.call(value, 'getRootNode') always is false
+            // MDN hasOwnProperty: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty
+            // hasOwnProperty returns a boolean indicating whether this object has the specified property as its own property
+            // getRootNode is inherited from the prototype chain, and defined on the Node prototype
             return (
                 value !== null &&
-                Object.prototype.hasOwnProperty.call(value, 'getRootNode') &&
+                'getRootNode' in value &&
```
Comment (IDisposable) [inline src/dom-to-image-more.js:673]: Nice

File: src/dom-to-image-more.js [hunk 3]
```diff
@@ -439,8 +439,14 @@
 
             function getRenderedChildren(original) {
                 if (util.isShadowSlotElement(original)) {
-                    return original.assignedNodes(); // shadow DOM <slot> has "assigned nodes" as rendered children
+                    const childElement = [
```
Comment (IDisposable) [inline src/dom-to-image-more.js:442]: So we're essentially `union`ing these two arrays... I assume there's no path by which the same nodes appear in both `childNodes` and `assignedNodes` (e.g. we're not going to generate duplicate nodes doing this.

**Review by IDisposable — CHANGES_REQUESTED** [review]

File: src/dom-to-image-more.js (hunk 3 above)
Comment (cWenyu) [reply src/dom-to-image-more.js:442]: You gave a good point here, we should not use the union, the assigned nodes actually will cover default content/node(original childNodes). Should use `or` relation here. I will update the code. Here's my test example:https://jsfiddle.net/CWenyu/wanphkx2/49/

Comment (cWenyu) [reply src/dom-to-image-more.js]: By the way, do you have suggestions on a starting point for the progress bar styling? I've been struggling with that for a few days 😵

**Comment by IDisposable** [comment]
Manually merged this in 3.4.0

**Pull request closed without merging**